security/sshdo/Portfile

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           makefile 1.0
+
+github.setup        raforg sshdo 1.1.1 v
+revision            0
+
+categories          security
+license             GPL-2+
+maintainers         {raf.org:raf @macportsraf} openmaintainer
+description         controls which commands may be executed via incoming ssh
+long_description    {*}${description}. \
+                    \n\nsshdo provides an easily configurable way of controlling which \
+                    commands may be executed via incoming ssh connections by specific \
+                    users and specific keys. Training mode allows a key to be used for \
+                    any command, while logging them. Then sshdo can learn from the logs \
+                    to know what commands to allow. It can also unlearn commands that \
+                    are no longer in use. This mitigates against private ssh key compromise \
+                    with very little effort.
+# Any of Python 2.6, 2.7, or 3.3+ will do.
+# When updating the python3 version, update the corresponding patch below as well.
+# Update the python3 version every year (or until there's a meta port:python3).
```

Comment:
There is no plan for there ever to be a meta python3 port. The `sudo port select` mechanism is the way that users would cause a `python3` symlink to exist.

You can make it easier to update the python version each year by setting it in a variable (well, two variables):

```
set python_branch       3.11
set python_version      [string map {. {}} ${python_branch}]
```

Then you only have to change a single line to update the python version.

In the patchfile, everywhere where you currently wrote `3.11` literally, instead use a placeholder like `@PYTHON_BRANCH@`. Then you can use the above variables wherever they're needed, e.g.:

```
depends_lib-append port:python${python_version}

post-patch {
    reinplace -W ${worksrcpath} "s|@PYTHON_BRANCH@|${python_branch}|g" Makefile sshdo test_sshdo
}
```

Reply:
Thanks. That's much better. It's done.